packages/dom-helper/lib/prop.js

```diff
@@ -13,11 +16,26 @@ export function normalizeProperty(element, attrName) {
     // TODO should this be an o_create kind of thing?
     cache = {};
     for (key in element) {
-      cache[key.toLowerCase()] = key;
+      if (isSettable(element, key)) {
+        cache[key.toLowerCase()] = key;
+      } else {
+        cache[key] = UNDEFINED;
+      }
     }
     propertyCaches[tagName] = cache;
   }
 
   // presumes that the attrName has been lowercased.
-  return cache[attrName];
+  var value = cache[attrName];
+  return value === UNDEFINED ? undefined : value;
+}
+
+function isSettable(element, attrName) {
+  var desc = getOwnPropertyDescriptor(element.constructor.prototype, attrName);
+  if (!desc) { return true; }
+  if (!desc.writable || !desc.hasOwnProperty('value') && typeof desc.set !== 'function') {
```

Comment:
Note, while writing the tests, I just realized the first part `!desc.writable` isn't specific enough because of implicit boolean conversion of `undefined`. it needs to be `desc.writable === false`